Virtual Librarian/Virtual Librarian/MainForm.cs

```diff
@@ -35,26 +35,41 @@ private void MainFormLoaded()
 
         private void DetectFaces(object sender, EventArgs e)
         {
-            Image<Bgr, Byte> currentFrame = capture.QueryFrame().ToImage<Bgr, Byte>();
+            try
+            {
+                Image<Bgr, Byte> currentFrame = capture.QueryFrame().ToImage<Bgr, Byte>();
+
+                if (currentFrame != null)
+                {
+                    Image<Gray, Byte> grayFrame = currentFrame.Convert<Gray, Byte>();
+                    // Face detections happens here by giving the classifier gray image
+                    detectedFaces = cascadeClassifier.DetectMultiScale(grayFrame, 1.1, 10, Size.Empty);
+
+                    foreach (var face in detectedFaces)
+                        // Draws rectangles in frame around faces
+                        currentFrame.Draw(face, new Bgr(0, double.MaxValue, 0), 3);
+
+                    //Returns the frame with drawn rectangles to live feed in Form
+                    imgCamUser.Image = currentFrame;
+                }
 
-            if (currentFrame != null)
+            }
+            catch
             {
-                Image<Gray, Byte> grayFrame = currentFrame.Convert<Gray, Byte>();
-                // Face detections happens here by giving the classifier gray image
-                detectedFaces = cascadeClassifier.DetectMultiScale(grayFrame, 1.1, 10, Size.Empty);
-
-                foreach (var face in detectedFaces)
-                    // Draws rectangles in frame around faces
-                    currentFrame.Draw(face, new Bgr(0, double.MaxValue, 0), 3);
-                //Returns the frame with drawn rectangles to live feed in Form
-                imgCamUser.Image = currentFrame;
+                this.Hide();
+                
+                if (MessageBox.Show("Camera not found", "Camera msg", MessageBoxButtons.OK) ==DialogResult.OK)
+                {
+                    this.Close();
+                   }
             }
         }
 
         private void LearnNewFace_Click(object sender, EventArgs e)
```

Comment:
consider renaming this function, and the button name  to something like ShowAvailableBooks (if I understand the context correctly)